### PR TITLE
Remove RestoreSources

### PR DESCRIPTION
**Pull request by tmat**
We decided to switch to using `NuGet.config` instead of `RestoreSources` msbuild property, since NuGet restore does not support authenticated feeds with the latter. 

To migrate, all repositories must follow steps:
1) Copy the list of feeds from `Version.props` file to `NuGet.config` file in the repository root. Keep the feeds in `Version.props` file.
2) Merge this PR and let the new build propagate to all repos.
3) Delete the code that sets `RestoreSources` in `Version.props` file in all repos (see e.g. https://github.com/dotnet/arcade/pull/3082)
4) Update source build infrastructure

  This change removes usage of `DotNetRestoreSources`, `DotNetRestoreSourcePropsPath` and `DotNetBuildOffline`.

### CODE REVIEW

**Comment by tmat** [comment]
@jcagme This will need the missing packages published.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props [hunk 1]
```diff
@@ -79,14 +68,14 @@
...
+    <MicrosoftNETTestSdkVersion Condition="'$(MicrosoftNETTestSdkVersion)' == ''">16.1.1</MicrosoftNETTestSdkVersion>
     <MicrosoftNetFrameworkReferenceAssembliesVersion Condition="'$(MicrosoftNetFrameworkReferenceAssembliesVersion)' == ''">1.0.0-preview.2</MicrosoftNetFrameworkReferenceAssembliesVersion>
     <MicrosoftVSSDKBuildToolsVersion Condition="'$(MicrosoftVSSDKBuildToolsVersion)' == ''">15.1.192</MicrosoftVSSDKBuildToolsVersion>
     <MicrosoftDiaSymReaderPdb2PdbVersion Condition="'$(MicrosoftDiaSymReaderPdb2PdbVersion)' == ''">1.1.0-beta1-62506-02</MicrosoftDiaSymReaderPdb2PdbVersion>
     <RoslynToolsNuGetRepackVersion Condition="'$(RoslynToolsNuGetRepackVersion)' == ''">1.0.0-beta3.18526.1</RoslynToolsNuGetRepackVersion>
     <MicrosoftDotNetSignToolVersion Condition="'$(MicrosoftDotNetSignToolVersion)' == ''">$(ArcadeSdkVersion)</MicrosoftDotNetSignToolVersion>
     <XliffTasksVersion Condition="'$(XliffTasksVersion)' == ''">1.0.0-beta.19252.1</XliffTasksVersion>
-    <XUnitVersion Condition="'$(XUnitVersion)' == ''">2.4.1-pre.build.4059</XUnitVersion>
+    <XUnitVersion Condition="'$(XUnitVersion)' == ''">2.4.1</XUnitVersion>
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: Please don't change this version as the newer stable package doesn't have a netstandard2.0 asset which will result in framework libraries being pulled into the output directory.

Comment (tmat) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: Is there a workitem tracking updating the release package?

Comment (ViktorHofer) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: We just discussed this in our infra stand-up. Currently in corefx conflict resolution doesn't work well for runtime libraries as we don't use the targeting pack Microsoft.NetCore.App.Refs. I created https://github.com/dotnet/sdk/issues/3335 today and @ericstj will help me figure out what we can do about this.

Comment (tmat) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props]: ok.

**Comment by tmat** [comment]
@chcosta @jcagme @markwilkie

**Comment by tmat** [comment]
@dagood FYI

File: eng/validate-sdk.ps1 [hunk 2]
```diff
@@ -87,8 +87,8 @@ try {
   CheckExitCode "Preserve artifacts for stage 1 build" $?
   Write-Host "Stage 2: Build using the local packages"
   
-  AddSourceToNugetConfig $nugetConfigPath $packagesSource
   CheckExitCode "Adding source to NuGet.config"
+  AddSourceToNugetConfig $nugetConfigPath $packagesSource
```
Comment (chcosta) [inline eng/validate-sdk.ps1]: Why did this get moved?

File: eng/validate-sdk.ps1 [hunk 3]
```diff
@@ -105,7 +105,7 @@ try {
   Write-Host "Building with updated dependencies"
 
   $ArtifactsLogDir = Join-Path (Join-Path $ArtifactsDir "log") $configuration
-  & .\common\cibuild.cmd -configuration $configuration @Args /p:AdditionalRestoreSources=$packagesSource /p:DotNetPublishBlobFeedUrl=https://dotnetfeed.blob.core.windows.net/dotnet-core-test/index.json
+  & .\common\cibuild.cmd -configuration $configuration @Args /p:DotNetPublishBlobFeedUrl=https://dotnetfeed.blob.core.windows.net/dotnet-core-test/index.json
```
Comment (chcosta) [inline eng/validate-sdk.ps1]: Is "dotnet-core-test" needed or just for testing?

**Comment by dagood** [comment]
FYI @crummel @dseefeld, looks like source-build will need to rely more on nuget.config mutation.

**Comment by crummel** [comment]
We will still need a way to override package versions - per https://github.com/dotnet/arcade/issues/2532 we had consolidated this all into `DotNetResourceSourcesProps`.  Can we add an import of `DotNetPackageVersionPropsPath`?

File: eng/validate-sdk.ps1 (hunk 2 above)
Comment (tmat) [reply eng/validate-sdk.ps1]: unintentional. will revert.

**Comment by jonfortescue** [comment]
@tmat @chcosta is this good to merge?

**Comment by tmat** [comment]
@jonfortescue No, step [1] above is not done.

**Comment by jonfortescue** [comment]
@riarenas @tmat what's the status on this?

**Comment by riarenas** [comment]
Same status as last time.

**Comment by jcagme** [comment]
This PR will be merged until all the repos have completed the 1st part. Status is tracked in https://github.com/dotnet/arcade/blob/master/Documentation/RestoreSourcesUpdateStatus.md

**Comment by ViktorHofer** [comment]
What about RestoreSources in Tools.props?

**Comment by tmat** [comment]
@ViktorHofer Those will be removed as well.

**Comment by ViktorHofer** [comment]
Great, I'll do the necessary changes in CoreFx.

**Comment by ViktorHofer** [comment]
Slightly unrelated but does sourcebuild already publish to the dotnet-core feed? Ideally we would also minimize the amount of feeds being used. Currently those are the additional feeds that we have in CoreFx:

https://github.com/dotnet/corefx/blob/master/eng/Tools.props#L3-L8

**Comment by crummel** [comment]
Source-build currently doesn't publish to any feeds, we only ship tarballs.  We'd be doing that in the far future if at all - we produce packages with the same version numbers as official packages but without all the runtimes that official packages have.  We don't want to poison anyone's NuGet caches, so when we do publish packages it will have to be to a separate feed just for us that isn't included by default anywhere.

**Comment by ViktorHofer** [comment]
SORRY 🤦‍♂️ sourceLINK not sourceBUILD

**Comment by tmat** [comment]
You shouldn't need sourcelink feed. The packages are on nuget.org

**Comment by ViktorHofer** [comment]
Thanks, removed it. What about `Microsoft.SymbolUploader.Build.Task`? Seems this is an arcade dependency, can we move it over to the dotnet-core feed?

**Comment by jcagme** [comment]
All repos in https://github.com/dotnet/arcade/blob/master/Documentation/RestoreSourcesUpdateStatus.md have been updated and now have restore sources in both, NuGet.config and Versions.props.

**Comment by riarenas** [comment]
@tmat Since this is a breaking change in the SDK, we will need to give heads up to everyone a couple of days before we plan to merge this just in case. I'll send the notice, and let's aim to merge this either Tuesday or Wednesday.

**Comment by tmat** [comment]
Is it a breaking change? As long as all the nuget.config files are updated nothing should break.

**Comment by tmat** [comment]
In any case, feel free to send emails and merge this whenever appropriate.

**Comment by chcosta** [comment]
Feels, to me, potentially disruptive enough that a heads up would be useful.  Thanks @tmat / @riarenas

**Comment by ViktorHofer** [comment]
It would be great if we could address my feedback and solve the conflicts. @tmat do you want to do it or should I?

**Comment by tmat** [comment]
@ViktorHofer Done. PTAL

**Comment by jcagme** [comment]
@tmat I'm planning to get this merged at 11AM, we are still good for that right?

**Pull request merged**